kyuubi-server/src/test/scala/org/apache/kyuubi/events/KyuubiEventLoggingServiceSuite.scala

```diff
@@ -82,7 +82,7 @@ class KyuubiEventLoggingServiceSuite extends WithKyuubiServer with HiveJDBCTestH
       val engineTable = engineStatementEventPath.getParent
       val resultSet2 = statement.executeQuery(s"SELECT * FROM `json`.`${engineTable}`" +
         "where statement = \"" + sql + "\"")
-      val engineStates = Array(INITIALIZED, PENDING, RUNNING, COMPILED, FINISHED)
+      val engineStates = Array(INITIALIZED, PENDING, RUNNING, COMPILED, COMPILED, FINISHED)
```

Comment:
I'm afraid the extra event. There is many redundant event info now ..

Reply:
There are indeed a lot of event logs. I read the spark engine. There are two events for starting the engine, two events for session opening and closing, and a new event for each statement execution state change, which is about 5 indivual.

My changes here will add an event to each statement.

Because the user's sql may not be executed quickly sometimes, the session page can better track the execution of a specific user's entire session.

I also planned to do one thing, like the spark thrift server, to list a list of jobids that are continuously generated by the statement, so that there are more events.

![image](https://user-images.githubusercontent.com/3898450/160158887-0c351ffe-b992-437a-ae6c-2ee2848d9992.png)

Reply:
after some thought, I think we can remove the existed `COMPILE`:
```SCALA
// TODO #921: COMPILED need consider eagerly executed commands
setState(OperationState.COMPILED)
```

then catch the event `SparkListenerSQLExecutionStart` in `SQLOperationListener`. At this time, we can post the `COMPILED` state event which contains the execution id.

This approach also fix the `TODO` about the eagerly executed commands.

Reply:
However, some DDLs do not generate jobs, so there is no executionId
e.g. use db / drop table.

But this TODO, I guess it is because the command such as insert select or ctas will be executed directly, so TODO wants to set this state in advance.

We can do this if we allow some DDL to have no `COMPILED` status.

Reply:
Now I think of a way, if there is no `COMPILED` state before setting the `FINISHED` state, then set the `COMPILED` state first.

Reply:
> However, some DDLs do not generate jobs, so there is no executionId

It's not right, query must have the executionId whether it generates jobs or not. So I said, we can catch  `SparkListenerSQLExecutionStart` rather than `SparkListenerJobStart`

Reply:
Sorry, after reading the Spark code, each SQL does have an executionId.
The DDL I watched directly from the kyuubi session page yesterday did not show the executionId.
That would also fix two issues by the way.
1. ddl shows executionId.
2. We can also remove listener when no job is generated to prevent memory leaks.

Reply:
After offline discussion, using `SparkListenerSQLExecutionStart` to get the `executionId`, it is difficult to map the current statement.
So ddl still can't display the `executionId`.
If there is no `executionId`, `SparkOperation#close` will remove the `SQLOperationListener` at the end, so there is no memory leak.